[2025-05-27T21:58:31.629+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: synthetic_data_generation.generate_supplier_data manual__2025-05-27T21:58:30.517421+00:00 [queued]>
[2025-05-27T21:58:31.634+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: synthetic_data_generation.generate_supplier_data manual__2025-05-27T21:58:30.517421+00:00 [queued]>
[2025-05-27T21:58:31.636+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 2
[2025-05-27T21:58:31.646+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): generate_supplier_data> on 2025-05-27 21:58:30.517421+00:00
[2025-05-27T21:58:31.649+0000] {standard_task_runner.py:57} INFO - Started process 96 to run task
[2025-05-27T21:58:31.652+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'synthetic_data_generation', 'generate_supplier_data', 'manual__2025-05-27T21:58:30.517421+00:00', '--job-id', '425', '--raw', '--subdir', 'DAGS_FOLDER/synthetic_data_dag.py', '--cfg-path', '/tmp/tmpbe3q5j9o']
[2025-05-27T21:58:31.654+0000] {standard_task_runner.py:85} INFO - Job 425: Subtask generate_supplier_data
[2025-05-27T21:58:31.687+0000] {task_command.py:416} INFO - Running <TaskInstance: synthetic_data_generation.generate_supplier_data manual__2025-05-27T21:58:30.517421+00:00 [running]> on host 7b83313548d6
[2025-05-27T21:58:32.094+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='synthetic_data_generation' AIRFLOW_CTX_TASK_ID='generate_supplier_data' AIRFLOW_CTX_EXECUTION_DATE='2025-05-27T21:58:30.517421+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-05-27T21:58:30.517421+00:00'
[2025-05-27T21:58:32.097+0000] {logging_mixin.py:154} INFO - Generating supplier data...
[2025-05-27T21:58:32.117+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1890, in _execute_context
    self.dialect.do_executemany(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 982, in do_executemany
    context._psycopg2_fetched_rows = xtras.execute_values(
  File "/home/airflow/.local/lib/python3.8/site-packages/psycopg2/extras.py", line 1299, in execute_values
    cur.execute(b''.join(parts))
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "ods_supplier_pkey"
DETAIL:  Key (supplier_id)=(1) already exists.


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/walmart-etl/generate_synthetic_data.py", line 90, in generate_supplier_data
    df_suppliers.to_sql('ods_supplier', engine, if_exists='append', index=False)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/core/generic.py", line 2878, in to_sql
    return sql.to_sql(
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 769, in to_sql
    return pandas_sql.to_sql(
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 1920, in to_sql
    total_inserted = sql_engine.insert_records(
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 1470, in insert_records
    raise err
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 1461, in insert_records
    return table.insert(chunksize=chunksize, method=method)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 1023, in insert
    num_inserted = exec_insert(conn, keys, chunk_iter)
  File "/home/airflow/.local/lib/python3.8/site-packages/pandas/io/sql.py", line 929, in _execute_insert
    result = conn.execute(self.table.insert(), data)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1385, in execute
    return meth(self, multiparams, params, _EMPTY_EXECUTION_OPTS)
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/sql/elements.py", line 334, in _execute_on_connection
    return connection._execute_clauseelement(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1577, in _execute_clauseelement
    ret = self._execute_context(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1953, in _execute_context
    self._handle_dbapi_exception(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 2134, in _handle_dbapi_exception
    util.raise_(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 211, in raise_
    raise exception
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1890, in _execute_context
    self.dialect.do_executemany(
  File "/home/airflow/.local/lib/python3.8/site-packages/sqlalchemy/dialects/postgresql/psycopg2.py", line 982, in do_executemany
    context._psycopg2_fetched_rows = xtras.execute_values(
  File "/home/airflow/.local/lib/python3.8/site-packages/psycopg2/extras.py", line 1299, in execute_values
    cur.execute(b''.join(parts))
sqlalchemy.exc.IntegrityError: (psycopg2.errors.UniqueViolation) duplicate key value violates unique constraint "ods_supplier_pkey"
DETAIL:  Key (supplier_id)=(1) already exists.

[SQL: INSERT INTO ods_supplier (supplier_id, supplier_name, contact_person, email, phone, address, city, state, zip_code, contract_start_date, source_system, load_timestamp) VALUES (%(supplier_id)s, %(supplier_name)s, %(contact_person)s, %(email)s, %(phone)s, %(address)s, %(city)s, %(state)s, %(zip_code)s, %(contract_start_date)s, %(source_system)s, %(load_timestamp)s)]
[parameters: ({'supplier_id': '1', 'supplier_name': 'Premier Logistics Solutions', 'contact_person': 'John Miller', 'email': 'john.miller@premierlogisticssolutions.com', 'phone': '+1-556-975-3908', 'address': '2597 Main St, Suite 548', 'city': 'New York', 'state': 'Texas', 'zip_code': '16962', 'contract_start_date': datetime.datetime(2010, 4, 18, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102555)}, {'supplier_id': '2', 'supplier_name': 'National Retail Inc', 'contact_person': 'Sarah Williams', 'email': 'sarah.williams@nationalretailinc.com', 'phone': '+1-923-245-7123', 'address': '5098 Main St, Suite 693', 'city': 'Chicago', 'state': 'North Carolina', 'zip_code': '15100', 'contract_start_date': datetime.datetime(2011, 7, 9, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102571)}, {'supplier_id': '3', 'supplier_name': 'Ultra Trading Industries', 'contact_person': 'Sarah Davis', 'email': 'sarah.davis@ultratradingindustries.com', 'phone': '+1-764-268-5350', 'address': '7238 Main St, Suite 901', 'city': 'Philadelphia', 'state': 'Pennsylvania', 'zip_code': '37997', 'contract_start_date': datetime.datetime(2011, 1, 25, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102580)}, {'supplier_id': '4', 'supplier_name': 'Value Supply Corp', 'contact_person': 'Mike Williams', 'email': 'mike.williams@valuesupplycorp.com', 'phone': '+1-964-230-5122', 'address': '6117 Main St, Suite 146', 'city': 'San Diego', 'state': 'Georgia', 'zip_code': '65013', 'contract_start_date': datetime.datetime(2012, 12, 7, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102589)}, {'supplier_id': '5', 'supplier_name': 'Premier Supply Inc', 'contact_person': 'David Jones', 'email': 'david.jones@premiersupplyinc.com', 'phone': '+1-724-875-5615', 'address': '4056 Main St, Suite 493', 'city': 'San Diego', 'state': 'New York', 'zip_code': '70604', 'contract_start_date': datetime.datetime(2011, 9, 25, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102597)}, {'supplier_id': '6', 'supplier_name': 'Elite Wholesale Group', 'contact_person': 'Lisa Davis', 'email': 'lisa.davis@elitewholesalegroup.com', 'phone': '+1-425-352-3969', 'address': '2725 Main St, Suite 212', 'city': 'San Diego', 'state': 'California', 'zip_code': '56920', 'contract_start_date': datetime.datetime(2011, 3, 24, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102605)}, {'supplier_id': '7', 'supplier_name': 'Superior Wholesale Inc', 'contact_person': 'Lisa Brown', 'email': 'lisa.brown@superiorwholesaleinc.com', 'phone': '+1-354-142-2179', 'address': '2395 Main St, Suite 888', 'city': 'San Antonio', 'state': 'North Carolina', 'zip_code': '17011', 'contract_start_date': datetime.datetime(2012, 10, 15, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102612)}, {'supplier_id': '8', 'supplier_name': 'Value Supply Enterprises', 'contact_person': 'David Smith', 'email': 'david.smith@valuesupplyenterprises.com', 'phone': '+1-520-933-4683', 'address': '1172 Main St, Suite 538', 'city': 'Philadelphia', 'state': 'Michigan', 'zip_code': '55189', 'contract_start_date': datetime.datetime(2012, 4, 23, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102620)}  ... displaying 10 of 50 total bound parameter sets ...  {'supplier_id': '49', 'supplier_name': 'Superior Merchandise Solutions', 'contact_person': 'Emily Johnson', 'email': 'emily.johnson@superiormerchandisesolutions.com', 'phone': '+1-631-288-9071', 'address': '7084 Main St, Suite 846', 'city': 'New York', 'state': 'California', 'zip_code': '53017', 'contract_start_date': datetime.datetime(2011, 3, 5, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102965)}, {'supplier_id': '50', 'supplier_name': 'National Merchandise Enterprises', 'contact_person': 'Emily Williams', 'email': 'emily.williams@nationalmerchandiseenterprises.com', 'phone': '+1-902-430-2325', 'address': '6122 Main St, Suite 349', 'city': 'Houston', 'state': 'Texas', 'zip_code': '12053', 'contract_start_date': datetime.datetime(2011, 6, 24, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102973)})]
(Background on this error at: https://sqlalche.me/e/14/gkpj)
[2025-05-27T21:58:32.138+0000] {taskinstance.py:1400} INFO - Marking task as UP_FOR_RETRY. dag_id=synthetic_data_generation, task_id=generate_supplier_data, execution_date=20250527T215830, start_date=20250527T215831, end_date=20250527T215832
[2025-05-27T21:58:32.152+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 425 for task generate_supplier_data ((psycopg2.errors.UniqueViolation) duplicate key value violates unique constraint "ods_supplier_pkey"
DETAIL:  Key (supplier_id)=(1) already exists.

[SQL: INSERT INTO ods_supplier (supplier_id, supplier_name, contact_person, email, phone, address, city, state, zip_code, contract_start_date, source_system, load_timestamp) VALUES (%(supplier_id)s, %(supplier_name)s, %(contact_person)s, %(email)s, %(phone)s, %(address)s, %(city)s, %(state)s, %(zip_code)s, %(contract_start_date)s, %(source_system)s, %(load_timestamp)s)]
[parameters: ({'supplier_id': '1', 'supplier_name': 'Premier Logistics Solutions', 'contact_person': 'John Miller', 'email': 'john.miller@premierlogisticssolutions.com', 'phone': '+1-556-975-3908', 'address': '2597 Main St, Suite 548', 'city': 'New York', 'state': 'Texas', 'zip_code': '16962', 'contract_start_date': datetime.datetime(2010, 4, 18, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102555)}, {'supplier_id': '2', 'supplier_name': 'National Retail Inc', 'contact_person': 'Sarah Williams', 'email': 'sarah.williams@nationalretailinc.com', 'phone': '+1-923-245-7123', 'address': '5098 Main St, Suite 693', 'city': 'Chicago', 'state': 'North Carolina', 'zip_code': '15100', 'contract_start_date': datetime.datetime(2011, 7, 9, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102571)}, {'supplier_id': '3', 'supplier_name': 'Ultra Trading Industries', 'contact_person': 'Sarah Davis', 'email': 'sarah.davis@ultratradingindustries.com', 'phone': '+1-764-268-5350', 'address': '7238 Main St, Suite 901', 'city': 'Philadelphia', 'state': 'Pennsylvania', 'zip_code': '37997', 'contract_start_date': datetime.datetime(2011, 1, 25, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102580)}, {'supplier_id': '4', 'supplier_name': 'Value Supply Corp', 'contact_person': 'Mike Williams', 'email': 'mike.williams@valuesupplycorp.com', 'phone': '+1-964-230-5122', 'address': '6117 Main St, Suite 146', 'city': 'San Diego', 'state': 'Georgia', 'zip_code': '65013', 'contract_start_date': datetime.datetime(2012, 12, 7, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102589)}, {'supplier_id': '5', 'supplier_name': 'Premier Supply Inc', 'contact_person': 'David Jones', 'email': 'david.jones@premiersupplyinc.com', 'phone': '+1-724-875-5615', 'address': '4056 Main St, Suite 493', 'city': 'San Diego', 'state': 'New York', 'zip_code': '70604', 'contract_start_date': datetime.datetime(2011, 9, 25, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102597)}, {'supplier_id': '6', 'supplier_name': 'Elite Wholesale Group', 'contact_person': 'Lisa Davis', 'email': 'lisa.davis@elitewholesalegroup.com', 'phone': '+1-425-352-3969', 'address': '2725 Main St, Suite 212', 'city': 'San Diego', 'state': 'California', 'zip_code': '56920', 'contract_start_date': datetime.datetime(2011, 3, 24, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102605)}, {'supplier_id': '7', 'supplier_name': 'Superior Wholesale Inc', 'contact_person': 'Lisa Brown', 'email': 'lisa.brown@superiorwholesaleinc.com', 'phone': '+1-354-142-2179', 'address': '2395 Main St, Suite 888', 'city': 'San Antonio', 'state': 'North Carolina', 'zip_code': '17011', 'contract_start_date': datetime.datetime(2012, 10, 15, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102612)}, {'supplier_id': '8', 'supplier_name': 'Value Supply Enterprises', 'contact_person': 'David Smith', 'email': 'david.smith@valuesupplyenterprises.com', 'phone': '+1-520-933-4683', 'address': '1172 Main St, Suite 538', 'city': 'Philadelphia', 'state': 'Michigan', 'zip_code': '55189', 'contract_start_date': datetime.datetime(2012, 4, 23, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102620)}  ... displaying 10 of 50 total bound parameter sets ...  {'supplier_id': '49', 'supplier_name': 'Superior Merchandise Solutions', 'contact_person': 'Emily Johnson', 'email': 'emily.johnson@superiormerchandisesolutions.com', 'phone': '+1-631-288-9071', 'address': '7084 Main St, Suite 846', 'city': 'New York', 'state': 'California', 'zip_code': '53017', 'contract_start_date': datetime.datetime(2011, 3, 5, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102965)}, {'supplier_id': '50', 'supplier_name': 'National Merchandise Enterprises', 'contact_person': 'Emily Williams', 'email': 'emily.williams@nationalmerchandiseenterprises.com', 'phone': '+1-902-430-2325', 'address': '6122 Main St, Suite 349', 'city': 'Houston', 'state': 'Texas', 'zip_code': '12053', 'contract_start_date': datetime.datetime(2011, 6, 24, 0, 0), 'source_system': 'Synthetic Data', 'load_timestamp': datetime.datetime(2025, 5, 27, 21, 58, 32, 102973)})]
(Background on this error at: https://sqlalche.me/e/14/gkpj); 96)
[2025-05-27T21:58:32.213+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-05-27T21:58:32.255+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
